.github/workflows/CI.yml

```diff
@@ -39,6 +39,97 @@ jobs:
         run: npx envinfo
       - name: Build
         run: ./configure && make -j8
+  build-tarball:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Environment Information
+        run: npx envinfo
+      - name: Make tarball
+        run: | 
+          export DISTTYPE=nightly
+          export DATESTRING=`date "+%Y-%m-%d"`
+          export COMMIT=xxxx
+          ./configure && make tar -j8
+          mkdir tarballs
+          mv *.tar.gz tarballs
+      - name: Upload tarball artifact
```

Comment:
This part should consider removed, see https://github.com/nodejs/node/pull/32458#issuecomment-603334891

Reply:
This step is needed so that downstream build jobs have access to the tarball.  I think this is a different case than https://github.com/nodejs/node/pull/32458#issuecomment-603334891 because it isn't an executable. The other option would be to make the tarball on every platform but it doesn't appear that it can be made on Windows as far as I can tell.  Also doing it that way would then make the jobs run longer as they would each have to build the tarball and then build from the tarball.